### PR TITLE
URL unreachable if not 200 HTTP status

### PR DESCRIPTION
**Pull request by marcreyesph**
Signed-off-by: Marc Reyes <marcreyesmedia@gmail.com>

@pombredanne, I added something to fetch.py (which is stated as TO DO in the said file) where the URL to be fetched is in 200 HTTP status or not. Returns True if the HTTP status is 200 and False if not.

### CODE REVIEW

File: src/commoncode/fetch.py [hunk 1]
```diff
@@ -76,11 +76,17 @@ def ping_url(url):
     Returns True is `url` is reachable.
     """
     import urllib2
-
-    # FIXME: if there is no 200 HTTP status, then the ULR may not be reachable.
+    # If there is no 200 HTTP status, then the URL may not be reachable.
+    http_url_error_list = [400,401,403,404,408,500,501,502,503,504,505]
```
Comment (pombredanne) [inline src/commoncode/fetch.py]: Can you make sure this follow PEP8?

File: src/commoncode/fetch.py [hunk 2]
```diff
@@ -76,11 +76,17 @@ def ping_url(url):
...
+    request = urllib2.Request(url)
     try:
-        urllib2.urlopen(url)
-    except Exception:
+        response = urllib2.urlopen(request)
+    except urllib2.HTTPError as e:
+        if e.code == any(http_url_error_list):
```
Comment (pombredanne) [inline src/commoncode/fetch.py]: any would return True or False.... How can this possibly work? Also, you would be better off with a set of return code IMHO.

**Comment by marcreyesph** [comment]
Thanks for the comments @pombredanne! I will apply necessary changes and commits to this pull request.

File: src/commoncode/fetch.py (hunk 2 above)
Comment (marcreyesph) [inline src/commoncode/fetch.py]: @pombredanne, what I did in the changes is for this to only have a single return code and print the HTTP response code as it returns a value of False.

File: src/commoncode/fetch.py [hunk 3]
```diff
@@ -39,6 +39,11 @@
 # logging.basicConfig(level=logging.DEBUG, stream=sys.stdout)
 # logger.setLevel(logging.DEBUG)
 
+def test_download(url):
```
Comment (pombredanne) [inline src/commoncode/fetch.py]: Would you mind call this check_download? Also what's the value to download and discard?

File: src/commoncode/fetch.py [hunk 4]
```diff
@@ -39,6 +39,11 @@
...
+    if ping_url(url) == True:
+        download_url(url, file_name=None, verify=True, timeout=10)
+    else:
+        print('Failed to start download for', url)
```
Comment (pombredanne) [inline src/commoncode/fetch.py]: You should raise an exception here instead

**Comment by marcreyesph** [comment]
Thanks, @pombredanne but I'm closing this pull request. I don't believe it has a purpose at this time. Thanks for the helping me in my proposal. Thank you very much!

**Pull request closed without merging**